lib/usd/translators/mayaReferenceUpdater.cpp

```diff
@@ -108,6 +112,49 @@ bool PxrUsdTranslators_MayaReferenceUpdater::shouldAutoEdit() const
 /* virtual */
 bool PxrUsdTranslators_MayaReferenceUpdater::canEditAsMaya() const { return true; }
 
+/* virtual */
+bool PxrUsdTranslators_MayaReferenceUpdater::editAsMaya()
+{
+    // Lock the transform node that corresponds to the Maya reference prim.
+    MDagPath transformPath;
+    if (MDagPath::getAPathTo(getMayaObject(), transformPath) != MS::kSuccess) {
+        return false;
+    }
+
+    MayaUsd::LockNodesUndoItem::lock(
+        "Maya reference pulled transform locking", transformPath, true);
+
+    // Lock all attributes except the transform attributes.  Create a set of
+    // Maya attributes that are converted to USD transform attributes.
+    // Children of compounds are ignored, as their parent will be locked.
+    static std::set<std::string> xformAttrNames;
+    if (xformAttrNames.empty()) {
+        for (const auto& opClass : UsdMayaXformStack::MayaStack().GetOps()) {
+            if (!opClass.IsInvertedTwin()) {
+                xformAttrNames.insert(std::string(opClass.GetName().GetText()));
+            }
+        }
+    }
+
+    MFnDependencyNode fn(getMayaObject());
+    auto              attrCount = fn.attributeCount();
+    for (unsigned int i = 0; i < attrCount; ++i) {
+        auto attr = fn.attribute(i);
+        auto plug = fn.findPlug(attr, true);
+        if (plug.isChild()) {
+            // Skip children of compounds, as their parent will be locked.
+            continue;
+        }
+        // If the attribute is not in our set of transform attributes, lock it.
+        MFnAttribute fnAttr(attr);
+        if (xformAttrNames.count(std::string(fnAttr.name().asChar())) == 0) {
+            plug.setLocked(true);
```

Comment:
As I understand, this part is not undoable directly? (I'll keep reading, maybe the undo is handled manually later.)

Reply:
Quite right...  An oversight on my part.  I don't see a way of performing this easily with OpenMaya; MDGModifier doesn't seem to have the required interface.  Suggestions welcome, otherwise I propose that we enter a ticket to do this work later.

Reply:
I think opening a ticket might be good at this point as we are nearing the 0.17 code complete date.